app/models/ability.rb

```diff
@@ -55,6 +56,35 @@ def initialize(member)
       end
       # note we don't support update for notifications
 
+      # can read seed requests if you are requester or seed owner
+      can :read,   SeedTrade, requester_id: member.id
+      can :read,   SeedTrade, seed: { :owner_id => member.id }
+      can :create, SeedTrade, requester_id: member.id
+
+      # you can't create a request for yourself
+      cannot :create, SeedTrade, seed: { :owner_id => member.id }
+
+      # you can decline if the trade was not accepted or declined
```

Comment:
This means that if you hit the wrong button by mistake, there's no way back, right? I can see social arguments either way - it means a trade is a binding commitment, but is quite hard on people with mobility issues who may hit wrong buttons quite often.